framework/src/base/UpdateDisplacedMeshThread.C

```diff
@@ -13,48 +13,56 @@
 /****************************************************************/
 
 #include "UpdateDisplacedMeshThread.h"
+
+#include "AllNodesSendListThread.h"
 #include "DisplacedProblem.h"
 #include "MooseMesh.h"
-#include "SubProblem.h"
+
+#include "libmesh/numeric_vector.h"
 
 UpdateDisplacedMeshThread::UpdateDisplacedMeshThread(FEProblemBase & fe_problem,
                                                      DisplacedProblem & displaced_problem)
-  : ThreadedNodeLoop<SemiLocalNodeRange, SemiLocalNodeRange::const_iterator>(fe_problem),
+  : ThreadedNodeLoop<NodeRange, NodeRange::const_iterator>(fe_problem),
     _displaced_problem(displaced_problem),
     _ref_mesh(_displaced_problem.refMesh()),
     _nl_soln(*_displaced_problem._nl_solution),
     _aux_soln(*_displaced_problem._aux_solution),
+    _nl_ghosted_soln(NumericVector<Number>::build(_nl_soln.comm()).release()),
```

Comment:
The first smart pointer is a UniquePtr, the second is a shared_ptr.

We can't make the shared_ptr into a unique_ptr, because we want to share these solutions between functors when the functors are split by multithreading.

Eventually we can directly construct the shared_ptr from unique_ptr, but that won't be guaranteed until after we euthanize pre-C++11 libMesh and get rid of the non-unique_ptr UniquePtr implementations.